src/main/MQ2Data.cpp

```diff
@@ -148,75 +148,31 @@ bool dataSubDefined(const char* szIndex, MQTypeVar& Ret)
 	return true;
 }
 
+// Arguments are passed in as Y, X, Z per documentation
 bool dataLineOfSight(const char* szIndex, MQTypeVar& Ret)
 {
 	if (!pControlledPlayer)
 		return false;
 
 	if (szIndex[0])
 	{
-		CVector3 SourcePos, DestPos;
-		SourcePos.X = DestPos.X = pControlledPlayer->Y;
-		SourcePos.Y = DestPos.Y = pControlledPlayer->X;
-		SourcePos.Z = DestPos.Z = pControlledPlayer->Z;
-
-		char szTemp[MAX_STRING];
-		strcpy_s(szTemp, szIndex);
-
-		// TODO:  This code appears in MQ2MathType Distance, possibly clean and combine
-		if (char* pColon = strchr(szTemp, ':'))
+		const auto args = split_view(szIndex, ':');
+		if (!args.empty())
 		{
-			*pColon = 0;
-			if (char* pComma = strchr(&pColon[1], ','))
-			{
-				*pComma = 0;
-				DestPos.X = GetFloatFromString(&pColon[1], DestPos.X);
-				*pComma = ',';
-				if (char* pComma2 = strchr(&pComma[1], ','))
-				{
-					*pComma2 = 0;
-					DestPos.Y = GetFloatFromString(&pComma[1], DestPos.Y);
-					*pComma2 = ',';
-					DestPos.Z = GetFloatFromString(&pComma2[1], DestPos.Z);
-				}
-				else
-				{
-					DestPos.Y = GetFloatFromString(&pComma[1], DestPos.Y);
-				}
-			}
-			else
-			{
-				DestPos.X = GetFloatFromString(&pColon[1], DestPos.X);
-			}
-		}
+			const glm::vec3 first_loc = GetVec3FromString(args[0], { pControlledPlayer->Y, pControlledPlayer->X, pControlledPlayer->Z });
+			glm::vec3 second_loc = { 0.0f, 0.0f, 0.0f };
```

Comment:
Old behavior if no second location is supplied defaulted to Player X Y Z.  new one seems to compare against 0,0,0 ?

Reply:
Good catch, but to confirm it was Y, X, Z, before yeah?  Can you check current changes and make sure I didn't mess that up?

Reply:
Confirmed.  Way it was done before was confusing with Y value in variable X.

Reply:
Thanks, just wanted to be sure.  I agree it was confusing before.  That's actually why I opted to use the coord1[0] notation instead of coord1.x (because the swap itself is already confusing)